src/Fantomas/FantomasConfig.fs

```diff
@@ -0,0 +1,146 @@
+          SemicolonAtEndOfLine   = cfg.SemicolonAtEndOfLine
+          SpaceBeforeArgument    = cfg.SpaceBeforeArgument
+          SpaceBeforeColon       = cfg.SpaceBeforeColon
+          SpaceAfterComma        = cfg.SpaceAfterComma
+          SpaceAfterSemicolon    = cfg.SpaceAfterSemicolon
+          IndentOnTryWith        = cfg.IndentOnTryWith
+          ReorderOpenDeclaration = cfg.ReorderOpenDeclaration
+          SpaceAroundDelimiter   = cfg.SpaceAroundDelimiter
+          StrictMode             = cfg.StrictMode }
+
+    /// The default configuration as a DTO
+    let internal defaultConfig() =
+        fromFormatConfig FormatConfig.Default
+
+type LoadedConfig =
+    {
+        Warnings: string[]
+        FileName: string
+        FantomasConfig: Dto.FantomasConfig
+        FormatConfig: FormatConfig
+    }
+
+/// The settings used for both serialization and deserialization
+let internal settings = 
+    JsonSerializerSettings(Formatting = Formatting.Indented, MissingMemberHandling = MissingMemberHandling.Error)
+
+/// Makes a new JsonSerializerSettings with warning output to the console for missing members
+let internal mkSettingsWithWarning missingMemberHandlerFn =
+    let settings = JsonSerializerSettings(Formatting = Formatting.Indented, MissingMemberHandling = MissingMemberHandling.Error)
+    settings.Error <- EventHandler<_>(fun _sender args -> 
+        match args.ErrorContext.Error.Message.Contains "Could not find member" with
+        | true -> 
+            missingMemberHandlerFn args
+            args.ErrorContext.Handled <- true
+        | false ->
+            ()
+    )
+    settings
+
+/// Serializes any object to a string
+let internal serialize o = 
+    JsonConvert.SerializeObject(o, settings)
+
+/// Deserializes any string to an object
+let internal deserialize<'t> json =
+    let warnings = ResizeArray<_>()
+    let res = JsonConvert.DeserializeObject<'t>(json, mkSettingsWithWarning (fun x -> warnings.Add x.ErrorContext.Error.Message))
+    res, warnings.ToArray()
+
+/// Populates (mutates) an object with the specified json and then returns the mutated object
+let internal populate<'t> (target: 't) json = 
+    let warnings = ResizeArray<_>()
+    JsonConvert.PopulateObject(json, target, mkSettingsWithWarning (fun x -> warnings.Add x.ErrorContext.Error.Message))
+    target, warnings.ToArray()
+
+/// Tries to load a config file from disk
+let tryLoad fileName =
+    try
+        match File.Exists fileName with
+        | true ->
+            fileName
+            |> File.ReadAllText 
+            |> populate (Dto.defaultConfig())
+            |> fun (cfg, warnings) -> 
+                { FileName = fileName
+                  Warnings = warnings
+                  FantomasConfig = cfg
+                  FormatConfig = Dto.toFormatConfig cfg }
+            |> Ok
+        | false -> 
+            Error(sprintf "File does not exist '%s'" fileName)
+    with ex -> 
+        Error(ex.Message)
```

Comment:
Why catch this if you are going to throw it anyway a few lines later?

Reply:
Both `load` and `tryLoad` have its place here, but I agree it would be better to have main body in `load` and call it from `tryLoad`.